### PR TITLE
More tests for the macros with generators

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/MArray.jl [hunk 1]
```diff
@@ -61,6 +61,7 @@
         @test (ex = macroexpand(:(@MArray fill)); isa(ex, Expr) && ex.head == :error)
         @test (ex = macroexpand(:(@MArray ones)); isa(ex, Expr) && ex.head == :error)
         @test (ex = macroexpand(:(@MArray sin(1:5))); isa(ex, Expr) && ex.head == :error)
+        @test (ex = macroexpand(:(@MArray si[])); isa(ex, Expr) && ex.head == :error)
```
Comment (c42f) [inline test/MArray.jl]: Why should this be an error?

At the moment this works fine:
```julia
julia> si = Int
Int64

julia> @SVector si[]
0-element SVector{0,Int64}
```

Comment (kshyatt) [inline test/MArray.jl]: Ah It wasn't working on latest master

**Comment by kshyatt** [comment]
I'll get rid of that last bad commit and then this should be ok

**Comment by c42f** [comment]
Looks good, thanks!

**Pull request merged**